packages/client/hmi-client/src/components/widgets/tera-input-field.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+	<div
+		class="input-container"
+		:class="{ error: errorMessage, disabled: disabled, focused: focused }"
+		@click="focusInput"
+	>
+		<label for="custom-input" class="input-label">{{ label }}</label>
+		<input
+			id="custom-input"
+			class="input-field"
+			ref="inputField"
+			:value="modelValue"
+			@input="updateValue"
+			:disabled="disabled"
+			@focus="onFocus"
+			@blur="onBlur"
+		/>
```

Comment:
@shawnyama @YohannParis @blanchco @Tom-Szendrey Tagging you'all as you have seen or involved with quirky input interactions in some manner.

We should come to some consensus no how a particular field should get updated. A pattern we have seen is that - - We use some computed-value to generate the UI, which includes input fields
- The input fields have listeners on @udpate events
- The listeners recalculates the computed-value, which regenerates the UI, which rerenders the input

So there are some problems here:
1. There are a lot of rendering cycles, this can be performance bottle neck 
2. The input is super quirky, because you type a character, and the focus is lost because the element itself is rerendered. So we are virtually entering 1-character-at-a-time

So we need some consensus on either/both
- How the UI is rendered
- How the inputbox changes are propagated, ie instead of @input directly maybe we need to add some debounce mechanism, or maybe we need to consider using @blur as well when focus is intentionally moved away.

Reply:
I dont mind have a meeting around this and coming to a consensus here, perhaps whatever we do decide we can add it in a followup PR?  As for how our custom input is updated here, I more or less follow how primevue is currently doing theirs. https://github.com/primefaces/primevue/blob/master/components/lib/inputtext/InputText.vue

Reply:
Yes, it will happen with this component. I can setup some thing to discuss.